packages/types/src/ComponentThemeVariables.ts

```diff
@@ -0,0 +1,1333 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015 - present Instructure, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import {
+  Border,
+  Breakpoints,
+  Forms,
+  Media,
+  Shadows,
+  Spacing,
+  Stacking,
+  Transitions,
+  Typography,
+  Size,
+  Radius,
+  StrokeWidth
+} from './BaseTheme'
+import { Colors } from './Colors'
+
+export type Avatar = {
+  color: Colors['textBrand']
+  background: Colors['backgroundLightest']
+  borderWidthSmall: Border['widthSmall']
+  borderWidthMedium: Border['widthMedium']
+  borderColor: Colors['borderMedium']
+  fontFamily: Typography['fontFamily']
+  fontWeight: Typography['fontWeightBold']
```

Comment:
Avatar got an update since, has a few extra theme variables, please add those too. Also, this is a good example why we should type the component theme variable object too, so the IDE can warn us:

```js
// Avatar theme.js
import { Avatar } from '@instructure/types'

const generateComponentTheme = (theme) => {
    const componentVariables: Avatar = { ... }
    ...
}
```

I know it will be a pain to add it to all components, but I can help with it if you want to. I think we should do it in this PR.

Reply:
Also, because of this usage, I would rename these types to "AvatarTheme" etc, just so it indicates theme variables, not props or sth else.

Reply:
Yeah, that is the reason I've made this "central" interface called `ThemeVariables` which contains all of the components theme variable typings, so one would use it like this: `ThemeVariables.Avatar` which I think is a bit less redundant than naming every component with <ComponentName>ThemeVariables.

Also I did not do the theme generator function typings on purpose in this PR, I it should be done in a different one, this PR is already pretty big.

Reply:
Cool, thanks! Tell me if you need help with batch-typing the generators :)

Reply:
I forgot to include the new Avatar theme variables, added it now

Reply:
Also, yeah it would be nice to do the batch-typing parallel :)